llarp/router/router.cpp

```diff
@@ -97,6 +97,84 @@ namespace llarp
     }
   }
 
+  util::StatusObject
+  Router::ExtractSummaryStatus() const
+  {
+    if (_running)
+    {
+      auto services = _hiddenServiceContext.ExtractStatus();
+      auto link_types = _linkManager.ExtractStatus();
+
+      int tx_rate = 0;
+      int rx_rate = 0;
+      int peers = 0;
+      for (const auto& links : link_types)
+      {
+        if (links == nlohmann::json::array())
+          continue;
+        for (const auto& link : links)
+        {
+          if (links == nlohmann::json::array())
+            continue;
+          const auto& established_peers = link["sessions"]["established"];
+          for (const auto& peer : established_peers)
+          {
+            tx_rate += peer["tx"].get<int>();
+            rx_rate += peer["rx"].get<int>();
+            peers++;
+          }
+        }
+      }
+
+      // Compute all stats on all path builders on the default endpoint
+      // Merge snodeSessions, remoteSessions and default into a single array
+      std::vector<nlohmann::json> builders;
+      auto snode_sessions = services["default"]["snodeSessions"];
+      for (const auto& session : snode_sessions)
```

Comment:
This is going to copy; either get as a reference, or just drop the temporary and use it directly in the `for` loop